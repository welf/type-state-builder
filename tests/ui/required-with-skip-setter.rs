use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct InvalidStruct {
    // This should be an error: required fields cannot skip setters
    #[builder(required, skip_setter)]
    name: String,

    email: Option<String>,
}

fn main() {}
