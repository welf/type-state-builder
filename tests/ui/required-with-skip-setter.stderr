error: Required fields cannot skip setters
       note: #[builder(skip_setter)] and #[builder(required)] are incompatible
       help: remove one of incompatible attributes
 --> tests/ui/required-with-skip-setter.rs:7:5
  |
7 |     name: String,
  |     ^^^^
